subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_connmon.c

```diff
@@ -125,7 +103,60 @@ static void modem_signal_update(struct k_work *work)
 	timestamp_prev = k_uptime_get_32();
 }
 
-int lwm2m_init_connmon(void)
+static void lwm2m_update_connmon_cell(void)
+{
+	k_work_submit(&modem_data_work);
+}
+
+static void lwm2m_update_connmon_mode(const enum lte_lc_lte_mode lte_mode)
+{
+	switch (lte_mode) {
+	case LTE_LC_LTE_MODE_LTEM:
+		lwm2m_engine_set_u8("4/0/0", LTE_FDD_BEARER);
+		break;
+
+	case LTE_LC_LTE_MODE_NBIOT:
+		lwm2m_engine_set_u8("4/0/0", NB_IOT_BEARER);
+		break;
+
+	case LTE_LC_LTE_MODE_NONE:
+	default:
+		LOG_DBG("No LTE mode information available");
+		break;
+	}
+}
+
+static void connmon_lte_notify_handler(const struct lte_lc_evt *const evt)
+{
+	int ret;
+	static bool connected;
+
+	switch (evt->type) {
+	case LTE_LC_EVT_CELL_UPDATE:
+		if (connected) {
+			lwm2m_update_connmon_cell();
+		}
+		break;
+
+	case LTE_LC_EVT_LTE_MODE_UPDATE:
+		lwm2m_update_connmon_mode(evt->lte_mode);
+		break;
+	case LTE_LC_EVT_NW_REG_STATUS:
+		connected = ((evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME) ||
+			    (evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_ROAMING));
+		if (connected) {
+			ret = modem_info_rsrp_register(modem_signal_handler);
```

Comment:
I think it should be enough to call this once in init? Or is there some reason why it can't be called there?

Also, I'm not sure if you really need to handle `LTE_LC_EVT_NW_REG_STATUS` at all. You could just call `lwm2m_update_connmon_cell()` always when `LTE_LC_EVT_CELL_UPDATE` is received. Yes, there would be some unnecessary updates (when out of service or in limited service), but I don't think that really matters.

Reply:
Yes there is a reason. If called in init, the rsrp register will fail. Also the `modem_info_params_get` fails as it doesn't get information about the current operator. I think that `modem_info_params_get` should try to get everything and not stop if it doesn't get some data.

Reply:
Ok, maybe you could add a comment in the code about this, because it isn't obvious.